ocaml/xapi/server_helpers.ml

```diff
@@ -164,9 +182,7 @@ let exec_with_forwarded_task ?http_other_config ?session_id ?origin task_id f =
     (fun ~__context -> f __context)
 
 let exec_with_subtask ~__context ?task_in_database ?task_description task_name f =
```

Comment:
It may be easier and cleaner to wrap the client call (on the master) in `exec_with_subtask`, and leave the server as it was.

Reply:
yes, that would be sensible

Reply:
though i need to make sure that if we fallback to the synchronous call, then we are not using a subtask